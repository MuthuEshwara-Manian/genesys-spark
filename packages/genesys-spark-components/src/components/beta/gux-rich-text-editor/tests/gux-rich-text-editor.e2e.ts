import { newSparkE2EPage, a11yCheck } from '../../../../test/e2eTestUtils';

describe('gux-rich-text-editor', () => {
  describe('#render', () => {
    [
      `<gux-rich-text-editor-beta>
  <gux-rich-text-editor-action-group slot="typographical-emphasis">
    <gux-rich-text-editor-action action="bold"></gux-rich-text-editor-action>
    <gux-rich-text-editor-action action="italic"></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="underline"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action action="strike"></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="text-styling">
    <gux-rich-text-editor-action
      action="clearFormatting"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="lists-indentation">
    <gux-rich-text-editor-action
      action="bulletList"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="orderedList"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="inserting">
    <gux-rich-text-editor-action
      action="codeblock"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="blockQuote"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action
    slot="global-action"
    action="delete"
  ></gux-rich-text-editor-action>
  <div class="editorElement" slot="editor"></div>
</gux-rich-text-editor-beta>`,
      `<gux-rich-text-editor-beta disabled>
  <gux-rich-text-editor-action-group slot="typographical-emphasis">
    <gux-rich-text-editor-action action="bold"></gux-rich-text-editor-action>
    <gux-rich-text-editor-action action="italic"></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="underline"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action action="strike"></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="text-styling">
    <gux-rich-text-editor-action
      action="clearFormatting"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="lists-indentation">
    <gux-rich-text-editor-action
      action="bulletList"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="orderedList"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action-group slot="inserting">
    <gux-rich-text-editor-action
      action="codeblock"
    ></gux-rich-text-editor-action>
    <gux-rich-text-editor-action
      action="blockQuote"
    ></gux-rich-text-editor-action>
  </gux-rich-text-editor-action-group>
  <gux-rich-text-editor-action
    slot="global-action"
    action="delete"
  ></gux-rich-text-editor-action>
  <div class="editorElement" slot="editor"></div>
</gux-rich-text-editor-beta>`
    ].forEach((html, index) => {
      it(`should display component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });

        const element = await page.find('gux-rich-text-editor-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });

  describe('a11y', () => {
    ['disabled'].forEach(property => {
      it(`should be accessible when "${property}" is set`, async () => {
        const html = `<gux-rich-text-editor-beta ${property}"></gux-rich-text-editor-beta>`;
        const page = await newSparkE2EPage({ html });

        await a11yCheck(page);
      });
    });
  });
});
