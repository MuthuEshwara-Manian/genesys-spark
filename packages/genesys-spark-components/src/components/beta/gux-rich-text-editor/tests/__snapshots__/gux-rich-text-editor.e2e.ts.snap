// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-rich-text-editor #render should display component as expected (1) 1`] = `"<gux-rich-text-editor-beta hydrated=""><template shadowrootmode="open"><div class="gux-rich-text-editor-container"><div class="gux-rich-text-editor-toolbar-container"><div class="gux-typographical-emphasis-container"><slot name="typographical-emphasis"></slot></div><div class="gux-text-styling-container"><slot name="text-styling"></slot></div><div class="gux-lists-indentation-container"><slot name="lists-indentation"></slot></div><div class="gux-inserting-container"><slot name="inserting"></slot></div><div class="gux-global-action-container"><slot name="global-action"></slot></div></div><slot name="editor"></slot></div></template> <gux-rich-text-editor-action-group slot="typographical-emphasis" hydrated=""> <gux-rich-text-editor-action action="bold" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="italic" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="underline" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="strike" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="text-styling" hydrated=""> <gux-rich-text-editor-action action="clearFormatting" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="lists-indentation" hydrated=""> <gux-rich-text-editor-action action="bulletList" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="orderedList" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="inserting" hydrated=""> <gux-rich-text-editor-action action="codeblock" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="blockQuote" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action slot="global-action" action="delete" hydrated=""></gux-rich-text-editor-action> <div class="editorElement" slot="editor"></div> </gux-rich-text-editor-beta>"`;

exports[`gux-rich-text-editor #render should display component as expected (2) 1`] = `"<gux-rich-text-editor-beta disabled="" hydrated=""><template shadowrootmode="open"><div class="gux-rich-text-editor-container gux-disabled"><div class="gux-rich-text-editor-toolbar-container"><div class="gux-typographical-emphasis-container"><slot name="typographical-emphasis"></slot></div><div class="gux-text-styling-container"><slot name="text-styling"></slot></div><div class="gux-lists-indentation-container"><slot name="lists-indentation"></slot></div><div class="gux-inserting-container"><slot name="inserting"></slot></div><div class="gux-global-action-container"><slot name="global-action"></slot></div></div><slot name="editor"></slot></div></template> <gux-rich-text-editor-action-group slot="typographical-emphasis" hydrated=""> <gux-rich-text-editor-action action="bold" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="italic" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="underline" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="strike" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="text-styling" hydrated=""> <gux-rich-text-editor-action action="clearFormatting" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="lists-indentation" hydrated=""> <gux-rich-text-editor-action action="bulletList" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="orderedList" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action-group slot="inserting" hydrated=""> <gux-rich-text-editor-action action="codeblock" hydrated=""></gux-rich-text-editor-action> <gux-rich-text-editor-action action="blockQuote" hydrated=""></gux-rich-text-editor-action> </gux-rich-text-editor-action-group> <gux-rich-text-editor-action slot="global-action" action="delete" hydrated=""></gux-rich-text-editor-action> <div class="editorElement" slot="editor"></div> </gux-rich-text-editor-beta>"`;
